### ISSUE TITLE
GCop628 false alarm

### ISSUE THREAD

**Issue by sungam3r** [Geeksltd/GCop#300]
```c#
public Task DoSome(int? value) // GCop628: Maybe define this method on 'Nullable' class as it's using 1 of its members (compared to 0 from this type)
{
    if (value.HasValue)
      return Task.CompletedTask;
    return null;
}
```
Obviously, we cannot move the method inside `System.Nullable<T>`. The rule should take into account the system types. Also note if you make the method static, then there is no warning.

**Comment by Foroughi** [comment]
@sungam3r Hi. Its weird b.c I'm not getting the same warning message on the same code. Here is the warning I'm getting : 

 GCop628: Maybe define this method on 'DoSome' class as it's using 1 of its members (compared to 0 from this type)

Would you please provide the entire class your working on?

**Comment by sungam3r** [comment]
```xml
    <PackageReference Include="GCop.All.Common" Version="2.6.0" PrivateAssets="all" />
```

```c#
using System.Threading.Tasks;

namespace Test
{
    internal class Class1
    {
        public Task DoSome(int? value) // GCop628: Maybe define this method on 'Nullable' class as it's using 1 of its members (compared to 0 from this type)
        {
            if (value.HasValue)
                return Task.CompletedTask;
            return null;
        }
    }
}
```

![image](https://user-images.githubusercontent.com/21261007/55896105-3a395980-5bc6-11e9-9066-5ff7c2bef348.png)

**Comment by Foroughi** [comment]
Even with using the exact version, I couldn't find this issue. Not in unit tests, not in a real scenario. 

Real scenario sample : 
![Screenshot (5)](https://user-images.githubusercontent.com/3727486/56965300-328c2580-6b72-11e9-993e-e2a656451494.png)

Unit test result :

![Screenshot (6)](https://user-images.githubusercontent.com/3727486/56965409-80a12900-6b72-11e9-954d-4681676d0bf2.png)

@marjanjavid @Karvan Would you please you guys have a look if you can find the issue on this one ?
@sungam3r Would you please share a light version of your repo?

**Comment by sungam3r** [comment]
[ConsoleApp7.zip](https://github.com/Geeksltd/GCop/files/3135288/ConsoleApp7.zip)
![изображение](https://user-images.githubusercontent.com/21261007/57030560-c848b300-6c4d-11e9-814e-2b0034ebe481.png)
![изображение](https://user-images.githubusercontent.com/21261007/57030585-e2829100-6c4d-11e9-8929-7773df5d1786.png)
